test/unit/vec/src/VecTest.cpp

```diff
@@ -468,3 +468,27 @@ TEST_CASE("accessByNameConstexpr", "[vec]")
     STATIC_REQUIRE(v4.z() == 3);
     STATIC_REQUIRE(v4.w() == 4);
 }
+
+TEMPLATE_TEST_CASE("Vec generator constructor", "[vec]", std::size_t, int, unsigned, float, double)
+{
+    // Define a generator function
+    auto generator = [](auto index) { return static_cast<TestType>(index.value + 1); };
+
+    // Create a Vec object using the generator function
+    alpaka::Vec<alpaka::DimInt<5>, TestType> vec(generator);
+
+    // Check that the values in the Vec object are as expected
+    for(std::size_t i = 0; i < 5; ++i)
+    {
+        // Floating point types require a precision check instead of an exact == match
+        if constexpr(std::is_floating_point<TestType>::value)
+        {
+            auto const precision = std::numeric_limits<TestType>::epsilon() * 5; // Arbitrary precision requirement
+            CHECK(std::abs(vec[i] - static_cast<TestType>(i + 1)) < precision);
+        }
+        else
+        {
+            CHECK(vec[i] == static_cast<TestType>(i + 1));
+        }
```

Comment:
Why do you need a comparison with an epsilon? All values fit within the representation of a `float`. Furthermore, if there were any rounding errors, they would be deterministic.

This should be enough:
```suggestion
            CHECK(vec[i] == static_cast<TestType>(i + 1));
```

Reply:
This will trigger a warning for unsafe == float comparison, and the test will fail because of that. An alternative would be to put a pragma disabling the warning, or to use `floatEqualExactNoWarning` which basically does exactly that

Reply:
Sigh ... alright then.

Reply:
Yeah... Here it's imho better option than disabling the pragmas that disable the warning - I'd prefer not to use them unless necessary, and it might be argued, that strict equality makes sense in test where the fp number is not being modified (allocations, initializations) whereas here you could imagine some future architecture, that would make the assignment i+1 imprecise, or someone reusing the test for bigger numbers. Not really a solid argument though, just trying to rationalize one of the options (eps comparison vs. pragma disabling the warning) since I don't like either.